### DISCUSSION TITLE
`getPayload` tries to deal with email even outside Next.js

### DISCUSSION THREAD

**Discussion by fofoy** [payloadcms/payload#10259] [Q&A]
### Describe the Bug

I reported some time ago about [a problem when using `getPayload` method outside of Next.js](https://github.com/payloadcms/payload/pull/9371) generating importMap outside Next.js when it was not needed and it was actually causing problems. This issue was kindly fixed by @r1tsuu but this didn't resolve the other issue I had which is that when I'm invoking `getPayload` method outside Next.js, it's trying to deal with some email stuff from my `payload.config.ts` that I'm importing.

Of course this is not needed at all when we just want to consume the Local API and generates errors in the console of the application that's consuming the Local API. Here is a sample log from my app :

![Image](https://github.com/user-attachments/assets/6687da1c-fad6-4238-8fe2-fdb10ad67650)

In a more general way, it seems like when we use Payload from outside Next.js, it's trying to do way more things that we'd like to, especially when you just want to consume the Local API. (e.g Sentry stuff).

![Image](https://github.com/user-attachments/assets/4a85bd38-b4dd-48d9-91c2-cc59d5c14ff1)

I don't know if something can be done to opt-out some plugins to be used only in Payload for example.

### Link to the code that reproduces this issue

https://github.com/fofoy/payload-getPayload-outside-nextjs

### Reproduction Steps

1 - Clone repo
2 - Run command `pnpm install`
3 - Run command `docker run -d --name payload-monorepo -p 27017:27017 mongo` to spin up a new MongoDB instance
4 - Run command `pnpm run dev` to start the apps in the monorepo
5 - Go to the next application URL, it should be `http://localhost:3001`
6 - Check out `next#dev` task and you should see following logs about email adapter not being provided

![Image](https://github.com/user-attachments/assets/4182abf5-b6c1-401d-8dc3-f4b098725243)

and also Sentry logs

![Image](https://github.com/user-attachments/assets/af8f4dc8-2644-461f-8048-de9e6abce819)

### Which area(s) are affected? (Select all that apply)

area: core, email-nodemailer, plugin: sentry

### Environment Info

```text
Binaries:
  Node: 20.12.2
  npm: 10.5.0
  Yarn: 1.22.19
  pnpm: 9.6.0
Relevant Packages:
  payload: 3.6.0
Operating System:
  Platform: linux
  Arch: x64
  Version: #1 SMP Tue Nov 5 00:21:55 UTC 2024
  Available memory (MB): 11967
  Available CPU cores: 8
```

**Comment by jmikrut** [comment]
Hey @fofoy — certain aspects of Payload's email service in particular are still very relevant in contexts outside of Next.js. For example, the Local API methods of `forgotPassword` and even the creation of a user with `auth.verify: true` will send emails. 

If you want to configure your own Payload instance to start up with specific things turned on / off, you can always do that with environment variables and by making your config dynamic, where it is constructed differently in certain contexts vs. others.

For example, with that approach, you could only add your email adapter when you want to. Same with the Sentry plugin.

I don't think there's much that we can do here though outside of that. I will convert this to a discussion though and will try and keep an eye on things as they progress here.

**Reply by fofoy** [reply]
Hey @jmikrut, thanks a lot for your feedback !
I like this approach and it makes sense, that could definitely work for me.
